generators/app/templates/src/StarterKit/Logging/HttpMessageLoggingHandler.cs

```diff
@@ -0,0 +1,102 @@
+    }
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
+      CancellationToken cancellationToken)
+    {
+      if (!_appSettings.RequestLogging.OutgoingEnabled)
+      {
+        return await base.SendAsync(request, cancellationToken);
+      }
+
+      var sw = new Stopwatch();
+      sw.Start();
+      if (request.Content != null && Log.IsEnabled(LogEventLevel.Debug))
+      {
+        string bodyAsText;
+        if (request.Content.Headers?.ContentType != null
+            && request.Content.Headers.ContentType.MediaType.ToLower().Contains("multipart/form-data"))
+        {
+          bodyAsText = "Logging of multipart content body disabled";
+        }
+        else
+        {
+          bodyAsText = await request.Content.ReadAsStringAsync();
+        }
+
+        Log.ForContext("Host", request.RequestUri.Host)
+          .ForContext("Headers", request.Headers)
+          .ForContext("Path", request.RequestUri.AbsolutePath)
+          .ForContext("Payload", bodyAsText)
+          .ForContext("Protocol", request.RequestUri.Scheme)
+          .ForContext("Method", request.Method)
+          .Information("API-call outgoing log Request");
+      }
+      else
+      {
+        Log.ForContext("Host", request.RequestUri.Host)
+          .ForContext("Headers", request.Headers)
+          .ForContext("Path", request.RequestUri.AbsolutePath)
+          .ForContext("Protocol", request.RequestUri.Scheme)
+          .ForContext("Method", request.Method)
+          .Information("API-call outgoing log Request");
+      }
+
+      HttpResponseMessage response = await base.SendAsync(request, cancellationToken);
+
+      sw.Stop();
+      if (response.Content != null &&
+          (Log.IsEnabled(LogEventLevel.Debug) || response.StatusCode.ToString().StartsWith("4")))
+      {
+        string bodyAsText;
+        if (response.Content.Headers?.ContentType != null
+            && response.Content.Headers.ContentType.MediaType.ToLower().Contains("multipart/form-data"))
+        {
+          bodyAsText = "Logging of multipart content body disabled";
+        }
+        else
+        {
+          bodyAsText = await response.Content.ReadAsStringAsync();
+        }
+
+        Log.ForContext("Headers", response.Headers)
+          .ForContext("Payload", bodyAsText)
+          .ForContext("Protocol", response.RequestMessage.RequestUri.Scheme)
+          .ForContext("Status", response.StatusCode)
+          .ForContext("Duration", sw.ElapsedMilliseconds)
+          .Information("API-call outgoing log Response");
+      }
+      else
+      {
+        Log.ForContext("Headers", response.Headers)
+          .ForContext("Protocol", response.RequestMessage.RequestUri.Scheme)
+          .ForContext("Status", response.StatusCode)
+          .ForContext("Duration", sw.ElapsedMilliseconds)
+          .Information("API-call outgoing log Response");
+      }
+
+
+      return response;
+    }
+  }
```

Comment:
Indien mogelijk zou de Logging-folder misschien nog verhuisd kunnen worden naar onder de "Framework"-folder. Dit is misschien wel eerder persoonlijke voorkeur om teveel folders op het "hoogste" niveau te vermijden qua overzichtelijkheid van de projectstructuur.